### PR TITLE
Support multiple clientid/username Qs params in clients API

### PR DESCRIPTION
**Pull request by SergeTupchiy**
Fixes EMQX-11900

Release version: v/e5.6

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [x] Added tests for the changes
- [x] Added property-based tests for code which performs user input validation
- [ ] Changed lines covered in coverage report
- [x] Change log has been added to `changes/(ce|ee)/(feat|perf|fix|breaking)-<PR-id>.en.md` files
- [x] For internal contributor: there is a jira ticket to track this change
- [ ] Created PR to [emqx-docs](https://github.com/emqx/emqx-docs) if documentation update is required, or link to a follow-up jira ticket
- [x] Schema changes are backward compatible

## Checklist for CI (.github/workflows) changes

- [ ] If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)
- [ ] Change log has been added to `changes/` dir for user-facing artifacts update

### CODE REVIEW

File: apps/emqx_management/src/emqx_mgmt_api_clients.erl [hunk 1]
```diff
@@ -656,6 +673,30 @@ fields(message) ->
         {from_clientid, hoconsc:mk(binary(), #{desc => ?DESC(msg_from_clientid)})},
         {from_username, hoconsc:mk(binary(), #{desc => ?DESC(msg_from_username)})},
         {payload, hoconsc:mk(binary(), #{desc => ?DESC(msg_payload)})}
+    ];
+fields(requested_client_fields) ->
+    %% NOTE: some Client fields actually returned in response are missing in schema:
+    %%  enable_authn, is_persistent, listener, peerport
```
Comment (SergeTupchiy) [inline apps/emqx_management/src/emqx_mgmt_api_clients.erl:679]: Should these fields be included to `client` schema? 
BTW, `peerport` is redundant as there is simply `port`.

File: changes/ce/feat-12719.en.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
+Support multiple clientid and username Query string parameters in "/clients" API.
+Add an option to specify which client info fields must be included in the response.
+
+Multi clientid/username queries examples:
+ - "/clients?clientid=client1&client=client2
```
Comment (zmstone) [inline changes/ce/feat-12719.en.md:5]:
```suggestion
 - "/clients?clientid=client1&clientid=client2
```

File: changes/ce/feat-12719.en.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+ - "/clients?username=user11&username=user2"
+ - "/clients?clientid=client1&client=client2&username=user1&username=user2"
```
Comment (zmstone) [inline changes/ce/feat-12719.en.md:7]:
```suggestion
 - "/clients?clientid=client1&clientid=client2&username=user1&username=user2"
```

File: apps/emqx_management/test/emqx_mgmt_api_clients_SUITE.erl [hunk 4]
```diff
@@ -715,6 +715,215 @@ t_query_clients_with_time(_) ->
     {ok, _} = emqx_mgmt_api_test_util:request_api(delete, Client1Path),
     {ok, _} = emqx_mgmt_api_test_util:request_api(delete, Client2Path).
 
+t_query_multiple_clients(_) ->
```
Comment (zmstone) [inline apps/emqx_management/test/emqx_mgmt_api_clients_SUITE.erl:718]: maybe add a test case to cover the kind of clientIDs which require url encoding.

File: changes/ce/feat-12719.en.md [hunk 5]
```diff
@@ -0,0 +1,11 @@
+Support multiple clientid and username Query string parameters in "/clients" API.
+Add an option to specify which client info fields must be included in the response.
```
Comment (zmstone) [inline changes/ce/feat-12719.en.md:2]: this is a great feature which deserves its own PR and release note ;)

File: apps/emqx_management/test/emqx_mgmt_api_clients_SUITE.erl (hunk 4 above)
Comment (SergeTupchiy) [reply apps/emqx_management/test/emqx_mgmt_api_clients_SUITE.erl:718]: Done

File: changes/ce/feat-12719.en.md (hunk 3 above)
Comment (SergeTupchiy) [reply changes/ce/feat-12719.en.md]: :crying_cat_face:  Thanks a lot, corrected.

File: changes/ce/feat-12719.en.md (hunk 5 above)
Comment (SergeTupchiy) [reply changes/ce/feat-12719.en.md]: Split into two blocks as discussed.

File: apps/emqx_management/src/emqx_mgmt_api_clients.erl (hunk 1 above)
Comment (zmstone) [reply apps/emqx_management/src/emqx_mgmt_api_clients.erl:679]: yes, make sense to add to schema

Comment (zmstone) [reply apps/emqx_management/src/emqx_mgmt_api_clients.erl:679]: maybe remove `port`

**Review by zmstone — APPROVED** [review]

**Comment by SergeTupchiy** [comment]
> maybe remove port

@zmstone , removing `port` would be a breaking change as this field is a part of the schema. Maybe better to remove undocumented `peerport` instead?

**Comment by zmstone** [comment]
> > @zmstone , removing `port` would be a breaking change as this field is a part of the schema. Maybe better to remove undocumented `peerport` instead?
> 
> both are breaking changes, maybe solve in 5.7

agree.
